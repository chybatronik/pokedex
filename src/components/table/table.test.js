import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import Table from './table'
import expect from 'expect'

Enzyme.configure({ adapter: new Adapter() })

const array = [
  { name: '1', isFetch: false, weight: 1, height: 2, sprites: {} },
  { name: '2', isFetch: false, weight: 21, height: 42, sprites: {} }
]

describe('table component', () => {
  it('renders call fun Table', () => {
    const mockCallback = jest.fn()
    const output = mount(
      <Table
        array={array}
        loadFunc={mockCallback}
        hasMore={true}
        isError={false}
      />
    )
    expect(mockCallback.mock.calls.length).toBe(1)
  })

  it('renders Table', () => {
    const mockCallback = jest.fn()
    const output = shallow(
      <Table
        array={array}
        loadFunc={mockCallback}
        hasMore={true}
        isError={false}
      />
    )
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  it('renders Table with error', () => {
    const mockCallback = jest.fn()
    const output = shallow(
      <Table
        array={array}
        loadFunc={mockCallback}
        hasMore={true}
        isError={true}
      />
    )
    expect(shallowToJson(output)).toMatchSnapshot()
  })
})
