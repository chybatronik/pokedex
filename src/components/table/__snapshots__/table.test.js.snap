// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table component renders Table 1`] = `
<InfiniteScroll
  element="article"
  hasMore={true}
  initialLoad={true}
  isReverse={false}
  loadMore={[MockFunction]}
  loader={
    <Row
      className="d-flex justify-content-center"
      tag="div"
    >
      <Spinner
        color="fuchsia"
        fadeIn="full"
        name="ball-scale-ripple"
        noFadeIn={false}
        overrideSpinnerClassName=""
      />
    </Row>
  }
  pageStart={0}
  ref={null}
  threshold={250}
  useCapture={false}
  useWindow={true}
>
  <Table
    responsiveTag="div"
    striped={true}
    tag="table"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          @
        </th>
        <th>
          name
        </th>
        <th>
          types
        </th>
        <th>
          weight
        </th>
        <th>
          height
        </th>
      </tr>
    </thead>
    <tbody>
      <RowComponent
        height={2}
        index={1}
        isFetch={false}
        key="row_1"
        name="1"
        sprites={Object {}}
        weight={1}
      />
      <RowComponent
        height={42}
        index={2}
        isFetch={false}
        key="row_2"
        name="2"
        sprites={Object {}}
        weight={21}
      />
    </tbody>
  </Table>
</InfiniteScroll>
`;

exports[`table component renders Table with error 1`] = `
<Alert
  closeAriaLabel="Close"
  color="danger"
  isOpen={true}
  style={
    Object {
      "width": "100%",
    }
  }
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  Ошибка подключения к серверу pokeapi.co
</Alert>
`;
